pkg/services/queryhistory/database.go

```diff
@@ -265,3 +266,48 @@ func (s QueryHistoryService) unstarQuery(ctx context.Context, user *models.Signe
 
 	return dto, nil
 }
+
+func (s QueryHistoryService) migrateQueries(ctx context.Context, user *models.SignedInUser, cmd MigrateQueriesToQueryHistoryCommand) (int, int, error) {
+	queryHistories := make([]*QueryHistory, 0, len(cmd.Queries))
+	starredQueries := make([]*QueryHistoryStar, 0)
+
+	err := s.SQLStore.WithTransactionalDbSession(ctx, func(session *sqlstore.DBSession) error {
+		for _, query := range cmd.Queries {
+			uid := util.GenerateShortUID()
+			queryHistories = append(queryHistories, &QueryHistory{
+				OrgID:         user.OrgId,
+				UID:           uid,
+				Queries:       query.Queries,
+				DatasourceUID: query.DatasourceUID,
+				CreatedBy:     user.UserId,
+				CreatedAt:     query.CreatedAt,
+				Comment:       query.Comment,
+			})
+
+			if query.Starred {
+				starredQueries = append(starredQueries, &QueryHistoryStar{
+					UserID:   user.UserId,
+					QueryUID: uid,
+				})
+			}
+		}
+
+		_, err := session.InsertMulti(queryHistories)
+		if err != nil {
+			return err
+		}
```

Comment:
nice! I would suggest though to run this in batches in case there are thousands of queries to migrate (the batch size is indicative; no need to use this one)
```suggestion
		batchsize := 50
		for i := 0; i < len(queryHistories); i += batchsize {
			j := i + batchsize
			if j > len(queryHistories) {
				j = len(queryHistories)
			}
			_, err := session.InsertMulti(queryHistories[i:j])
			if err != nil {
				return err
			}
		}
```

Same for the `starredQueries`.